compiler/src/dotty/tools/dotc/core/classfile/ClassfileParser.scala

```diff
@@ -62,8 +67,6 @@ class ClassfileParser(
   import ClassfileConstants._
   import ClassfileParser._
 
-  protected val in: AbstractFileReader = new AbstractFileReader(classfile)
```

Comment:
Since we're only using one DataReader now, we could restore the `in` field.

Reply:
This is intentional, as passing it around (rather than storing it in fields) is a more principled way of using resources. With the cool analysis @abgruszecki is developing, we might be able to have formal guarantees.